pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/BrokersBase.java

```diff
@@ -437,5 +440,79 @@ private void doShutDownBrokerGracefully(int maxConcurrentUnloadPerSec,
         pulsar().getBrokerService().unloadNamespaceBundlesGracefully(maxConcurrentUnloadPerSec, forcedTerminateTopic);
         pulsar().closeAsync();
     }
+
+    /**
+     * dynamically update log4j2 logger level at runtime.
+     *
+     * @param targetClassName : class name to update
+     * @param targetLevel     : target log level
+     */
+    private CompletableFuture<Void> internalUpdateLoggerLevelAsync(String targetClassName, String targetLevel) {
+        CompletableFuture<Void> loggerLevelFuture = new CompletableFuture<>();
+        CompletableFuture.runAsync(() -> {
```

Comment:
Why are we executing this a a unknown thread?
We can execute this code in the main thread of the http handler

Reply:
You're right. updated.